hollow/src/main/java/com/netflix/hollow/core/index/key/HollowPrimaryKeyValueDeriver.java

```diff
@@ -144,11 +146,20 @@ private Object readValue(int ordinal, int fieldIdx) {
         int lastFieldPath = fieldPathIndexes[fieldIdx].length - 1;
         for (int i = 0; i < lastFieldPath; i++) {
             int fieldPosition = fieldPathIndexes[fieldIdx][i];
+            if (ordinal == ORDINAL_NONE) {
```

Comment:
This seemed like a logical place to add null checks, since `HollowPrimaryKeyValueDeriver` supports getting the resolved pkey value. But this is also used for indexing. Should the logic to resolve a pkey, while handling null fields exist here or somewhere else?

Reply:
Not feeling great about introducing branching in the happy path, for the validator use case. Maybe for the validator it would be better to iterate over the fields in a primary key, then for each field iterate over the ordinals and call read ordinal at every part of the field path to check if we get an ORDINAL_NONE back?